sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamTableFunctionScanRel.java

```diff
@@ -95,29 +186,19 @@ public TableFunctionScan copy(
           input);
       String operatorName = ((RexCall) getCall()).getOperator().getName();
       checkArgument(
-          operatorName.equals("TUMBLE"),
-          "Only support TUMBLE table-valued function. Current operator: %s",
+          TVFStreamingUtils.WINDOWING_TVF.contains(operatorName),
+          "Only support %s table-valued functions. Current operator: %s",
+          TVFStreamingUtils.WINDOWING_TVF,
           operatorName);
-      RexCall call = ((RexCall) getCall());
-      RexInputRef wmCol = (RexInputRef) call.getOperands().get(1);
-      PCollection<Row> upstream = input.get(0);
-      Schema outputSchema = CalciteUtils.toSchema(getRowType());
-      FixedWindows windowFn = FixedWindows.of(durationParameter(call.getOperands().get(2)));
-      PCollection<Row> streamWithWindowMetadata =
-          upstream
-              .apply(ParDo.of(new FixedWindowDoFn(windowFn, wmCol.getIndex(), outputSchema)))
-              .setRowSchema(outputSchema);
-
-      PCollection<Row> windowedStream =
-          assignTimestampsAndWindow(
-              streamWithWindowMetadata, wmCol.getIndex(), (WindowFn) windowFn);
 
-      return windowedStream;
+      return tvfToPTransformMap.get(operatorName).toPTransform(((RexCall) getCall()), input.get(0));
```

Comment:
I am planning to keep it for now. Because we will support user-defined table-valued function in the near future, seems to me that such way will be extensible for UDTVF.

Reply:
If it turns that UDTVF will goes to different Rel or same Rel but different code path, I will switch here back to 'switch case'